### PR TITLE
feat: adds cms-payload starter

### PR DESCRIPTION
**Pull request by JessChowdhury**
### Description

New example to provide a NextJS starter built with Payload, which can be deployed together in a single instance.

### Type of Change

- [x] New Example
- [ ] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

### New Example Checklist

- [ ] 🛫 `npm run new-example` was used to create the example
- [x] 📚 The template wasn't used but I carefuly read the [Adding a new example](https://github.com/vercel/examples#adding-a-new-example) steps and implemented them in the example
- [x] 📱 Is it responsive? Are mobile and tablets considered?

### Additional Comments

I added the front matter metadata to allow this example to be featured as a template.

Thank you 🙏

### CODE REVIEW

File: starter/cms-payload/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+name: Payload with Next.js Starter
+slug: cms-payload
+description: Next.js app with Payload, all in one repo.
+framework: Next.js
+useCase:
+  - Starter
+css: CSS Modules
+deployUrl: https://vercel.com/new/git/external?repository-url=https://github.com/vercel/examples/tree/main/starter/cms-payload&project-name=cms-payload&repository-name=cms-payload
+demoUrl: https://cms-payload.vercel.app
+relatedTemplates:
```
Comment (leerob) [inline starter/cms-payload/README.md:11]: I believe this is required, could we add some? Maybe just the Next.js boilerplate starter.

File: starter/cms-payload/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+name: Payload with Next.js Starter
+slug: cms-payload
+description: Next.js app with Payload, all in one repo.
+framework: Next.js
+useCase:
+  - Starter
+css: CSS Modules
+deployUrl: https://vercel.com/new/git/external?repository-url=https://github.com/vercel/examples/tree/main/starter/cms-payload&project-name=cms-payload&repository-name=cms-payload
+demoUrl: https://cms-payload.vercel.app
```
Comment (leerob) [inline starter/cms-payload/README.md:10]: 404

File: starter/cms-payload/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+relatedTemplates:
+---
+
+# Next + Payload Serverless Demo
+
+This example shows how to utilize `@payloadcms/next-payload` to deploy Payload serverlessly, in the same repo alongside of a NextJS app.
```
Comment (leerob) [inline starter/cms-payload/README.md:16]:
```suggestion
This example shows how to utilize `@payloadcms/next-payload` to deploy Payload serverless, in the same repo alongside of a Next.js app.
```

File: starter/cms-payload/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## Demo
+
+https://cms-payload.vercel.app
+
+## Deploy your own
+
+Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example):
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/examples/tree/main/starter/cms-payload&project-name=cms-payload&repository-name=cms-payload)
```
Comment (leerob) [inline starter/cms-payload/README.md:26]: Does this work without providing env vars?

File: starter/cms-payload/app/(site)/[slug]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      },
+    },
+  })
+
+  const page = pages.docs[0]
+
+  if (!page) return notFound()
+
+  return (
+    <React.Fragment>
```
Comment (leerob) [inline starter/cms-payload/app/(site)/[slug]/page.tsx:25]: This can probably just be `<> <>`

File: starter/cms-payload/app/head.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export default function Head() {
```
Comment (leerob) [inline starter/cms-payload/app/head.tsx:1]: Could you migrate this to the Metadata API? 

https://nextjs.org/docs/app/building-your-application/optimizing/metadata

File: starter/cms-payload/app/layout.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  return (
+    <html>
+      <head />
```
Comment (leerob) [inline starter/cms-payload/app/layout.tsx:8]:
```suggestion
```

File: starter/cms-payload/payload/fields/richText/index.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    leaves: [],
+  }
+) =>
+  deepMerge<RichTextField, Partial<RichTextField>>(
+    {
+      name: 'richText',
+      type: 'richText',
+      required: true,
+      admin: {
+        // upload: {
```
Comment (leerob) [inline starter/cms-payload/payload/fields/richText/index.ts:32]: Is this needed?

File: starter/cms-payload/vercel.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+  "rewrites": [{ "source": "/admin/(.*)", "destination": "/admin/index.html" }],
```
Comment (leerob) [inline starter/cms-payload/vercel.json:2]: This should use `next.config.js` (also you can cut the github silent config?)

File: starter/cms-payload/README.md (hunk 4 above)
Comment (JessChowdhury) [reply starter/cms-payload/README.md:29]: Only `MONGODB_URI` and `PAYLOAD_SECRET` are required - I updated the `.env.example` file to clarify.

**Review by leerob — APPROVED** [review]

**Pull request merged**